src/Microsoft.AspNetCore.Mvc.Core/Internal/RedirectToActionResultExecutor.cs

```diff
@@ -46,7 +47,24 @@ public void Execute(ActionContext context, RedirectToActionResult result)
             }
 
             _logger.RedirectToActionResultExecuting(destinationUrl);
-            context.HttpContext.Response.Redirect(destinationUrl, result.Permanent);
+            if (result.PreserveMethod == false)
+            {
+                context.HttpContext.Response.Redirect(destinationUrl, result.Permanent);
+            }
+
+            else
+            {
```

Comment:
Should this whole block of logic just be a new extension method in https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNetCore.Http.Extensions/ResponseExtensions.cs to hang off of https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNetCore.Http.Abstractions/HttpResponse.cs?

We're not doing anything terribly special here - I could imagine apps wanting this functionality as well.

@Tratcher thoughts on adding some convenience extension overloads/methods to `Response.Redirect()` that support "Preserve Method" semantics? (HTTP 307/308)